### DISCUSSION TITLE
Windows wheels depending on libgcc_s_seh-1.dll and libstdc++-6.dll, why? How to use clang-cl?

### DISCUSSION THREAD

**Discussion by paugier** [pypa/cibuildwheel#1798] [Q&A]
I try to build wheels for Windows with cibuildwheel for a project containing C++ extensions (https://github.com/fluiddyn/fluidimage/). The C++ files are produced with Pythran (a Python compiler also used by Scipy) and this project uses Meson (also like Scipy). The build passes (see for example https://github.com/fluiddyn/fluidimage/actions/runs/8455184364/job/23162048001) and wheels are uploaded on PyPI. However, the extensions cannot be imported because the .pyd files depend on libgcc_s_seh-1.dll and libstdc++-6.dll.

```
Dump of file .\Lib\site-packages\fluidimage\calcul\__pythran__\correl.cp311-win_amd64.pyd

File Type: DLL

  Image has the following dependencies:

    libgcc_s_seh-1.dll
    KERNEL32.dll
    api-ms-win-crt-environment-l1-1-0.dll
    api-ms-win-crt-heap-l1-1-0.dll
    api-ms-win-crt-math-l1-1-0.dll
    api-ms-win-crt-private-l1-1-0.dll
    api-ms-win-crt-runtime-l1-1-0.dll
    api-ms-win-crt-stdio-l1-1-0.dll
    api-ms-win-crt-string-l1-1-0.dll
    api-ms-win-crt-time-l1-1-0.dll
    libstdc++-6.dll
    python311.dll
```

I don't understand why there are these dependencies on Windows. It seems to indicate that GCC is used (through Mingw?).

I know that to build this project on Windows, we need to use clang-cl but I didn't find in the doc how to do that with cibuildwheel.

CI config: https://github.com/fluiddyn/fluidimage/blob/branch/default/.github/workflows/wheels.yml

**Comment by henryiii** [comment]
Quick thought: have you tried the windows wheel repair tool - delvewheel? The example for using it is in the docs. https://cibuildwheel.pypa.io/en/stable/options/#repair-wheel-command

**Reply by henryiii** [reply]
Also, meson is rather picky sometimes, make sure you’ve also tried activating MSVC to see if that helps. You could compare with how SciPy builds their wheels.
